tools/dify_neo4j.bak/provider/neo4j_query.py

```
@@ -0,0 +1,19 @@
+
+class Neo4jQueryProvider(ToolProvider):
+    def _validate_credentials(self, credentials: dict[str, Any]) -> None:
+        url = credentials.get("AURA_URL")
+        username = credentials.get("AURA_USERNAME")
+        password = credentials.get("AURA_PASSWORD")
+        if not all([url, username, password]):
+            raise ToolProviderCredentialValidationError("All Neo4j credentials must be provided.")
+        try:
+            with GraphDatabase.driver(url, auth=basic_auth(username, password)) as driver:
+                driver.verify_connectivity()
+        except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)